doc/manual/manual.tex

```diff
@@ -1572,6 +1572,145 @@ \subsubsection{Free surface stabilization}
+The equations are solved for the solid velocity $\mathbf{u_s}$, the fluid pressure $p_f$, and an additional 
+variable, the compaction pressure $p_c$, which is related to the fluid and solid pressure through the relation 
+$p_c = (1-\phi) (p_s-p_f)$. $K_D$ is the Darcy coefficient, which is defined as the quotient of the permeability 
+and the fluid viscosity and $\Gamma$ is the melting rate. $\eta$ and $\xi$ are the shear and compaction viscosities 
+and can depend on the porosity, temperature, pressure, strain rate and composition. However, there are various 
+laws for these quantities and so they are implemented in the material model. Common formulations for the dependence 
+on porosity are $\eta = (1-\phi) \eta_0 e^{-\alpha_\phi \phi}$ with $\alpha_\phi \approx 25...30$ and 
+$\xi = \eta_0 \phi^{-n}$ with $n \approx 1$.
+
+To avoid the density gradients in Equation \ref{eq:stokes-2-melt}, which would have to be specified individually 
+for each material model by the user, we can use the same method as for the mass conservation (described in section 
+\ref{sec:boussinesq}) and assume the change in solid density is dominated by the change in static pressure, 
+which can be written as
+$\nabla p_s \approx \nabla p_\text{static} \approx \rho_s \textbf{g}$.
+This finally allows us to write
+\begin{equation*}
+\frac{1}{\rho_s} \nabla \rho_s
+\approx \frac{1}{\rho_s} \frac{\partial \rho_s}{\partial p_s} \nabla p_s
+\approx \frac{1}{\rho_s} \frac{\partial \rho_s}{\partial p_s} \nabla p_s
+\approx \frac{1}{\rho_s} \frac{\partial \rho_s}{\partial p_s} \rho_s \textbf{g}
+\approx \kappa_s \rho_s \textbf{g}. 
+\end{equation*}
+For the fluid pressure, choosing a good approximation depends on the model parameters and setup (see \cite{DH2015}). 
+Hence, we make $\nabla \rho_{f}$ a model input parameter, which can be adapted based on the forces that are expected 
+to be dominant in the model. 
+We then can replace the second equation by
+\begin{align*}
+\nabla \cdot \mathbf u_s - \nabla \cdot K_D \nabla p_f 
+  - K_D \nabla p_f \cdot \frac{\nabla \rho_f}{\rho_f}
+  &= 
+  - \nabla \cdot (K_D\rho_f \mathbf g)
+  \\
+  &\quad
+  + \Gamma \left( \frac{1}{\rho_f} - \frac{1}{\rho_s} \right)
+  \notag
+  \\
+  &\quad
+  - \frac{\phi }{\rho_f} \mathbf u_s \cdot \nabla\rho_f
+  - (\mathbf u_s \cdot \mathbf g ) (1 - \phi) \kappa_s \rho_s
+  \notag
+  \\
+  &\quad
+  - K_D \mathbf g \cdot \nabla \rho_f .
+  \notag
+\end{align*}
+%
+The melt velocity is computed as
+\[
+ \mathbf u_f =  \mathbf u_s - \frac{K_D}{\phi} (\nabla p_f - \rho_f g),
+\]
+but is only used for postprocessing purposes and for computing the time step length.  
+
+\note{Here, we do not use the visco-elasto-plastic rheology of the \cite{KMK2013} formulation. 
+Hence, we do not consider the elastic deformation terms that would appear on the right hand side of Equation 
+\ref{eq:stokes-1-melt} and Equation \ref{eq:stokes-3-melt} and that include the elastic and compaction stress 
```

Comment:
Here and elsewhere, make sure the space between `Equation` and `\ref` is a no-break space `~`.

Oh, also, please use `\eqref` instead of `\ref` to make sure there are parentheses around the eq. number.